Inital commit for Hamster Time Tracker ⏰

rabin-io opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented:
This replaces #2614 that was closed automatically.

hfiguiere commented on .github/workflows/flatpak-builder.yml:
```diff
@@ -0,0 +1,82 @@
+---
+
+name: Pre-Flathub PR Workflow
```
not sure why you need this.

FWIW flathub repositories get automatically build on new PR and on commits into the relevant branches.

rabin-io replied:
Mostly for my self, to shorten the feedback loop, and see this my PR will fail for other test.

barthalion replied:
Please remove it. We have our own CI.

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,52 @@
+  - '--socket=fallback-x11'
+  - '--share=ipc'
+  - '--filesystem=xdg-documents'
+  - '--own-name=org.gnome.Hamster'
```
```suggestion
```
You don't need this it's implicit with the app-id

rabin-io replied:
removed

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,52 @@
+---
+app-id: org.gnome.Hamster
```
It this a GNOME hosted project ?

(changing the app-id has consquences all down the way, including with DBus, so I'm not requesting it)

rabin-io replied:
I'm not sure, I used the same ID in the XML they have in there own repo.

nedrichards replied:
Yeah this is one of those older apps from before GNOME Circle etc.

hfiguiere commented:
bot, build org.gnome.Hamster

hfiguiere commented:
seems it can't find the appdata.

nedrichards commented:
bot, build org.gnome.Hamster

rabin-io commented:
> seems it can't find the appdata.

I'm having hard time to "install" this file, even though I created it in the repo, and help please ?

nedrichards commented:
Sure, just had a quick look. The main thing is that you don't install it like this https://github.com/rabin-io/hamster/commit/8cb0ed6e62ec420922633d67479607ae10691718 you install it like the metainfo file above. The issue with that file is that it's not using the declared app-id so it doesn't get automatically compiled. Should you actually be using `org.gnome.Hamster.GUI`?

rabin-io commented:
> Sure, just had a quick look. The main thing is that you don't install it like this [rabin-io/hamster@8cb0ed6](https://github.com/rabin-io/hamster/commit/8cb0ed6e62ec420922633d67479607ae10691718) that's the fully compiled version you install it like the metainfo file above and then the build system compiles it along with the desktop files and icons to make the .gz compressed version. The issue with that file is that it's not using the declared app-id so it doesn't get automatically compiled. Should you actually be using `org.gnome.Hamster.GUI`?

I really not sure, the "GUI" is just one portion of the application, and I don't why they split it into 2 app id's. What do you think I should do ?

Also, I was able to add the XML files via the flatpak build process, so I think I will drop the install via the `wscript`.

what do you think ?

nedrichards commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,53 @@
+    sources:
+      - type: archive
+        url: >-
+          https://github.com/rabin-io/hamster/archive/refs/heads/master.zip
```
I'm going to reask the question from the previous PR: 

Is there a reason this doesn't point to the upstream source release but to your fork instead?

nedrichards commented:
There are ways of exporting multiple 'apps' out of a single flatpak but I'm not sure they're particularly useful in this case. Also, overriding an upstream selected app-id almost always goes badly. If the app wants to be `org.gnome.Hamster.GUI` then why not let it?

nedrichards commented:
bot, build org.gnome.Hamster

(you can run that command as well btw. when you push a new version)

rabin-io commented:
bot, build org.gnome.Hamster

rabin-io commented:
bot, build org.gnome.Hamster

rabin-io commented:
Why the build failed ?

bot, build org.gnome.Hamster

pretty please :)

barthalion commented:
Sounds like a problem with the appdata/metainfo file.

```
No valid root node specified
```

barthalion commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,54 @@
+  - '--share=ipc'
+  - '--filesystem=xdg-documents'
+modules:
+  - name: intltool
```
Please switch to intltool from https://github.com/flathub/shared-modules/tree/master/intltool.

barthalion commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,54 @@
+      - ./waf configure --prefix=${FLATPAK_DEST}
+      - ./waf build
+      - ./waf install
+      - install -vD --mode 644 data/org.gnome.Hamster.GUI.metainfo.xml /app/share/app-info/xmls/org.gnome.Hamster.xml
```
This is wrong. You shouldn't place any files in `app-info` manually.

barthalion commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,54 @@
+      - ./waf build
+      - ./waf install
+      - install -vD --mode 644 data/org.gnome.Hamster.GUI.metainfo.xml /app/share/app-info/xmls/org.gnome.Hamster.xml
+      - gzip /app/share/app-info/xmls/org.gnome.Hamster.xml
```
Same as above.

barthalion commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,54 @@
+      - ./waf install
+      - install -vD --mode 644 data/org.gnome.Hamster.GUI.metainfo.xml /app/share/app-info/xmls/org.gnome.Hamster.xml
+      - gzip /app/share/app-info/xmls/org.gnome.Hamster.xml
+      - install -vD --mode 644 data/org.gnome.Hamster.GUI.metainfo.xml /app/share/appdata/org.gnome.Hamster.xml
```
```suggestion
      - install -vD --mode 644 data/org.gnome.Hamster.GUI.metainfo.xml /app/share/metainfo/org.gnome.Hamster.metainfo.xml
```

rabin-io commented:
bot, build org.gnome.Hamster

rabin-io commented:
I submitted a [PR](https://github.com/projecthamster/hamster/pull/690) upstream to resolve the problem with the missing tags.

hfiguiere commented:
you can add the patch here. It's ok to carry a patch to fix this until it's fixed in a release upstream.

rabin-io commented:
bot, build org.gnome.Hamster

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,50 @@
+    builddir: true
+    prefix: /app
+    build-commands:
+      - patch -p1 -i 00-fix-missing-tags.patch
```
1. that's not how you apply patches
2. the patch is not even in this module.

```suggestion
```

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,50 @@
+        url: >-
+          https://files.pythonhosted.org/packages/ea/13/de39ddf4f9f9cea0c7684cd54a50d79c97ea99c9f6aed798fd13d0bd4609/pyxdg-0.27-py2.py3-none-any.whl
+        sha256: 2d6701ab7c74bbab8caa6a95e0a0a129b1643cf6c298bf7c569adec06d0709a0
+      - type: patch
+        path: 00-fix-missing-tags.patch
```
```suggestion
```

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,50 @@
+      - type: archive
+        url: >-
+          https://github.com/projecthamster/hamster/archive/refs/tags/v3.0.2.tar.gz
+        sha256: 8bf28fe1f06cd950583ebfb24acf5254d3406ee29c3046f6a6fb648b98c929ef
```
```suggestion
        sha256: 8bf28fe1f06cd950583ebfb24acf5254d3406ee29c3046f6a6fb648b98c929ef
     - type: patch
        path: 00-fix-missing-tags.patch
```

hfiguiere replied:
the indentation might be off here. :-/

hfiguiere commented:
You didn't add the patch where I suggested to do.

rabin-io commented:
> You didn't add the patch where I suggested to do.

I'm sorry, I don't understand  that do you meant by "add here", I was assuming next to the yaml build file. So were exactly ?

hfiguiere commented:
> > You didn't add the patch where I suggested to do.
> 
> I'm sorry, I don't understand that do you meant by "add here", I was assuming next to the yaml build file. So were exactly ?

THIS change https://github.com/flathub/flathub/pull/2636#discussion_r751616426

It was simply ignored.

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,49 @@
+        url: >-
+          https://files.pythonhosted.org/packages/ea/13/de39ddf4f9f9cea0c7684cd54a50d79c97ea99c9f6aed798fd13d0bd4609/pyxdg-0.27-py2.py3-none-any.whl
+        sha256: 2d6701ab7c74bbab8caa6a95e0a0a129b1643cf6c298bf7c569adec06d0709a0
+      - type: patch
+        path: patches/00-fix-missing-tags.patch
```
This is not the right place. It should be down after line 49.

hfiguiere replied:
See previously:
https://github.com/flathub/flathub/pull/2636#discussion_r751616426

rabin-io commented:
> > > You didn't add the patch where I suggested to do.
> > 
> > 
> > I'm sorry, I don't understand that do you meant by "add here", I was assuming next to the yaml build file. So were exactly ?
> 
> THIS change [#2636 (comment)](https://github.com/flathub/flathub/pull/2636#discussion_r751616426)
> 
> It was simply ignored.

Not ignoring.... just didn't understand you at first 😇

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,49 @@
+sdk: org.gnome.Sdk
+command: hamster
+rename-appdata-file: org.gnome.Hamster.GUI.metainfo.xml
+rename-desktop-file: org.gnome.Hamster.GUI.desktop
```
```suggestion
rename-desktop-file: org.gnome.Hamster.GUI.desktop
rename-icon: org.gnome.Hamster.GUI
```

That's why it can't find the icon

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,49 @@
+    sources:
+      - type: archive
+        url: >-
+          https://github.com/projecthamster/hamster/archive/refs/heads/master.zip
```
the buid will fail as soon as a new commit is pushed to `master`.

rabin-io commented:
As for using the master branch, not sure if I should go with the very old release 3.0.2, which require more patch's, or use the per-released 3.0.3 branch. (I personally build and use from master, as this project move very slowly)

rabin-io commented:
bot, build org.gnome.Hamster

rabin-io commented:
🎉🎉🎉 Congratulation ... 

Now what ?

hfiguiere commented:
> 🎉🎉🎉 Congratulation ...
> 
> Now what ?

This: https://github.com/flathub/flathub/pull/2636/files#r751800594

you need a specific revision.

barthalion commented on patches/01-add-remote-icon.patch:
```diff
@@ -0,0 +1,12 @@
+   <name>Hamster</name>
+   <summary>Your personal time keeping tool</summary>
+   <content_rating type="oars-1.1" />
++  <icon type="remote">https://raw.githubusercontent.com/projecthamster/hamster/master/data/art/scalable/org.gnome.Hamster.GUI.svg</icon>
```
Any reason to use a remote icon here, instead of shipping SVG with the app?

barthalion commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,52 @@
+    sources:
+      - type: archive
+        url: >-
+          https://github.com/projecthamster/hamster/archive/refs/heads/master.zip
```
As @hfiguiere says, please change it to specific commit.

barthalion commented:
Have you contacted upstream developers about this submission?

rabin-io commented:
Yes, https://github.com/projecthamster/hamster/issues/686
But there are not very responsive. I can change the url to point to the current 3.0.3 branch which is there, but not yet released, but that will have the same problem as with master, when a new commit will be add.

What do you think ?

hfiguiere commented on org.gnome.Hamster.yaml:
```diff
@@ -0,0 +1,52 @@
+      - ./waf install
+      - install -vD --mode 644 data/org.gnome.Hamster.GUI.metainfo.xml ${FLATPAK_DEST}/share/metainfo/org.gnome.Hamster.metainfo.xml
+    sources:
+      - type: archive
+        url: >-
+          https://github.com/projecthamster/hamster/archive/refs/heads/release-3.0.3.zip
+        sha256: a3c9740802ad40d3cb3b0917d8105fda7e80cb128784e1081a0ebd69466f553e
```
```suggestion
      - type: git
        url: https://github.com/projecthamster/hamster.git
        commit: 70e2547e7b155bda76ceeeb92f26af5bc177fb0a
```

This is how you should do it, by using git instead of the snapshot.

hfiguiere commented:
bot, build org.gnome.Hamster

nedrichards commented:
Crashes on launch for me:

> [nedr@nedr-fedora ~]$ flatpak run org.gnome.Hamster
Traceback (most recent call last):
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 177, in activate_name_owner
    return self.get_name_owner(bus_name)
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 361, in get_name_owner
    return self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/app/lib/python3.9/site-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: The name does not have an owner

> During handling of the above exception, another exception occurred:

> Traceback (most recent call last):
  File "/app/bin/hamster", line 149, in on_activate_window
    self._open_window(action.get_name(), data)
  File "/app/bin/hamster", line 184, in _open_window
    self.overview_controller = Overview()
  File "/app/lib/python3.9/site-packages/hamster/overview.py", line 477, in __init__
    self.find_facts()
  File "/app/lib/python3.9/site-packages/hamster/overview.py", line 531, in find_facts
    self.facts = self.storage.get_facts(start, end, search_terms=search)
  File "/app/lib/python3.9/site-packages/hamster/client.py", line 156, in get_facts
    for fact in self.conn.GetFactsJSON(dbus_range, search_terms)]
  File "/app/lib/python3.9/site-packages/hamster/client.py", line 96, in conn
    self._connection = dbus.Interface(self.bus.get_object('org.gnome.Hamster',
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 241, in get_object
    return self.ProxyObjectClass(self, bus_name, object_path,
  File "/app/lib/python3.9/site-packages/dbus/proxies.py", line 250, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 182, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 277, in start_service_by_name
    return (True, self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/app/lib/python3.9/site-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
Traceback (most recent call last):
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 177, in activate_name_owner
    return self.get_name_owner(bus_name)
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 361, in get_name_owner
    return self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/app/lib/python3.9/site-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NameHasNoOwner: The name does not have an owner

> During handling of the above exception, another exception occurred:

> Traceback (most recent call last):
  File "/app/bin/hamster", line 149, in on_activate_window
    self._open_window(action.get_name(), data)
  File "/app/bin/hamster", line 184, in _open_window
    self.overview_controller = Overview()
  File "/app/lib/python3.9/site-packages/hamster/overview.py", line 477, in __init__
    self.find_facts()
  File "/app/lib/python3.9/site-packages/hamster/overview.py", line 531, in find_facts
    self.facts = self.storage.get_facts(start, end, search_terms=search)
  File "/app/lib/python3.9/site-packages/hamster/client.py", line 156, in get_facts
    for fact in self.conn.GetFactsJSON(dbus_range, search_terms)]
  File "/app/lib/python3.9/site-packages/hamster/client.py", line 96, in conn
    self._connection = dbus.Interface(self.bus.get_object('org.gnome.Hamster',
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 241, in get_object
    return self.ProxyObjectClass(self, bus_name, object_path,
  File "/app/lib/python3.9/site-packages/dbus/proxies.py", line 250, in __init__
    self._named_service = conn.activate_name_owner(bus_name)
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 182, in activate_name_owner
    self.start_service_by_name(bus_name)
  File "/app/lib/python3.9/site-packages/dbus/bus.py", line 277, in start_service_by_name
    return (True, self.call_blocking(BUS_DAEMON_NAME, BUS_DAEMON_PATH,
  File "/app/lib/python3.9/site-packages/dbus/connection.py", line 652, in call_blocking
    reply_message = self.send_message_with_reply_and_block(
dbus.exceptions.DBusException: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable

rabin-io commented:
Missing permission for DBUS ?

nedrichards commented:
Quite possibly, I assume it runs fine for you when installed from the test remote? Certainly it should be run with the --log-session-bus argument to see what finish args it actually requires.

rabin-io commented:
I did run for me, here is the dump when enabling `--log-session-bus`, I really don't know how to read that.

```
❯ flatpak run --log-session-bus  org.gnome.Hamster
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.gnome.Hamster
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.gnome.Hamster
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.gnome.Hamster
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C10
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C11
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C12
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C13
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C14
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.gnome.Hamster
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.gnome.Hamster
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.gnome.Hamster
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 2)
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C10
B-1: <- org.freedesktop.DBus return from C11
*REWRITTEN*
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
Filtering message due to arg0 org.gtk.vfs.Daemon, policy: 0 (required 1)
B-1: <- org.freedesktop.DBus return from C12
*REWRITTEN*
C13: -> org.gtk.vfs.Daemon call org.gtk.vfs.MountTracker.ListMountableInfo at /org/gtk/vfs/mounttracker
*HIDDEN* (ping)
B-1: <- org.freedesktop.DBus return from C13
*REWRITTEN*
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C15
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C16
C17: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C17
C18: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C18
C19: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C19
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.gnome.Hamster
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.gnome.Hamster
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.gnome.Hamster
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
C10: -> org.freedesktop.DBus call org.freedesktop.DBus.RequestName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C10
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
C11: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C12: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C13: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C14: -> org.freedesktop.DBus call org.freedesktop.DBus.StartServiceByName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C11
B-1: <- org.freedesktop.DBus return from C12
B-1: <- org.freedesktop.DBus return from C13
B-1: <- org.freedesktop.DBus return from C14
C15: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C15
C16: -> :1.350 call org.freedesktop.DBus.Properties.GetAll at /org/freedesktop/portal/desktop
B376: <- :1.350 return from C16
C17: -> :1.350 call org.freedesktop.portal.Inhibit.CreateMonitor at /org/freedesktop/portal/desktop
C20: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C20
C21: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C21
C22: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C22
C23: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C23
C24: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C24
B389: <- :1.350 return from C17
B390: <- :1.350 signal org.freedesktop.portal.Request.Response at /org/freedesktop/portal/desktop/request/1_75803/t

(org.gnome.Hamster.GUI:2): Gtk-WARNING **: 20:33:45.921: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
C1: -> org.freedesktop.DBus call org.freedesktop.DBus.Hello at /org/freedesktop/DBus
C2: -> org.freedesktop.DBus fake wildcarded AddMatch for org.freedesktop.portal
C3: -> org.freedesktop.DBus fake wildcarded AddMatch for org.gnome.Hamster
C4: -> org.freedesktop.DBus fake wildcarded AddMatch for org.mpris.MediaPlayer2.org.gnome.Hamster
C5: -> org.freedesktop.DBus fake ListNames
B-1: <- org.freedesktop.DBus return from C1
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameAcquired at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C2
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C3
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C4
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C5
C6: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.DBus
C7: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Desktop
C8: -> org.freedesktop.DBus fake GetNameOwner for org.freedesktop.portal.Documents
C9: -> org.freedesktop.DBus fake GetNameOwner for org.gnome.Hamster
C10: -> org.freedesktop.DBus fake GetNameOwner for org.gnome.Hamster.GUI
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C6
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C7
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C8
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C9
*SKIPPED*
B-1: <- org.freedesktop.DBus return from C10
*SKIPPED*
C25: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C25
C26: -> :1.75789 call org.freedesktop.DBus.Introspectable.Introspect at /org/gnome/Hamster
B208: <- :1.75789 return from C26
C27: -> :1.75789 call org.gnome.Hamster.Version at /org/gnome/Hamster
B209: <- :1.75789 return from C27
C28: -> :1.75789 call org.gnome.Hamster.GetFactsJSON at /org/gnome/Hamster
B210: <- :1.75789 return from C28
C18: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
C19: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C20: -> org.freedesktop.DBus call org.freedesktop.DBus.GetNameOwner at /org/freedesktop/DBus
C11: -> org.freedesktop.portal.IBus call org.freedesktop.DBus.Peer.Ping at /org/freedesktop/IBus
B-1: <- org.freedesktop.DBus return from C18
B-1: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NameHasNoOwner from C19
B-1: <- org.freedesktop.DBus return error org.freedesktop.DBus.Error.NameHasNoOwner from C20
C21: -> org.freedesktop.portal.Desktop call org.freedesktop.portal.Session.Close at /org/freedesktop/portal/desktop/session/1_75803/gtk52142295
C22: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C23: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C24: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C25: -> org.freedesktop.DBus call org.freedesktop.DBus.ReleaseName at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus return from C22
B-1: <- org.freedesktop.DBus return from C23
B-1: <- org.freedesktop.DBus return from C24
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameLost at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
B-1: <- org.freedesktop.DBus signal org.freedesktop.DBus.NameOwnerChanged at /org/freedesktop/DBus
```

barthalion commented:
bot, build org.gnome.Hamster

barthalion commented:
@nedrichards It actually starts for me? Mind trying again with the current build?

nedrichards commented:
/merge

Works fine now - clearly some sort of transient woo on my developer system - apologies for the false alarm.
